### PR TITLE
Backlink from individual referral

### PR DESCRIPTION
**Pull request by patinipat**
# Context

[Trello Card](https://trello.com/c/kHs8Hej1/1432-a-referrer-can-navigate-back-to-their-referral-dashboard-from-an-individual-referral)

# Changes in this PR
Add a back link in the show referral page

### Before
image

### After
image

# Release checklist

[Release process documentation](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/edit-v2/4247847062?draftShareId=a1c360ab-bd31-4db1-aae3-7cc002761de9).

## Pre-merge checklist

- [ ] Are any changes required to dependent services for this change to work?
  (eg. CAS API)
- [ ] Have they been released to production already?

## Post-merge checklist

- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to test
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to preprod
- [ ] [Manually approve](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4247847062/Release+process#Manual-releases)
  release to prod

### CODE REVIEW

**Comment by patinipat** [comment]
[A referrer can navigate back to their referral dashboard from an individual referral](https://trello.com/c/kHs8Hej1/1432-a-referrer-can-navigate-back-to-their-referral-dashboard-from-an-individual-referral)

File: integration_tests/tests/apply/show.cy.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  it('navigates back from the show application page to the list applications page', () => {
+    // Given there are applications in the database
+    const inProgressApplications = applicationFactory.buildList(3, { status: 'inProgress' })
+    const firstApplication = inProgressApplications[0]
+
+    cy.task('stubApplications', inProgressApplications)
+    cy.task('stubApplicationGet', { application: firstApplication })
+
+    // Given I am signed in
+    cy.signIn()
```
Comment (bryangaledxw) [inline integration_tests/tests/apply/show.cy.ts:22]: I'd suggest move this to the top of the test, think that's what we've got most other places

File: cypress_shared/pages/apply/show.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { TemporaryAccommodationApplication as Application } from '../../../server/@types/shared'
+import paths from '../../../server/paths/apply'
+import Page from '../page'
+
+export default class ShowPage extends Page {
```
Comment (bryangaledxw) [inline cypress_shared/pages/apply/show.ts]: Don't think we should need this - we have `TaskListPage` for this page, though you'll need to give it a static visit method

File: cypress_shared/pages/apply/show.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  static visit(inProgressApplication: Application): ShowPage {
+    cy.visit(paths.applications.show({ id: inProgressApplication.id }))
+
+    return new ShowPage(inProgressApplication)
+  }
+
+  clickBack(): void {
+    cy.get('a').contains('Back').click()
+  }
```
Comment (bryangaledxw) [inline cypress_shared/pages/apply/show.ts]: We've a `clickBack()` method on the base class, so we shouldn't need to have add one here

**Review by bryangaledxw — APPROVED** [review]
Looks good, nice first PR. Couple of comments but otherwise this is good to go in. Shout if you want me to take a look again before you merge

File: cypress_shared/pages/apply/show.ts (hunk 3 above)
Comment (patinipat) [reply cypress_shared/pages/apply/show.ts]: :recycle:

File: cypress_shared/pages/apply/show.ts (hunk 2 above)
Comment (patinipat) [reply cypress_shared/pages/apply/show.ts]: will the the ``TaskListPage instead

File: cypress_shared/helpers/apply.ts [hunk 4]
```diff
@@ -313,7 +313,7 @@ export default class ApplyHelper {
 
   completeSentenceInformation() {
     // Given I click the sentence information task
-    Page.verifyOnPage(TaskListPage).clickTask('sentence-information')
+    Page.verifyOnPage(TaskListPage, this.application).clickTask('sentence-information')
```
Comment (tahb) [inline cypress_shared/helpers/apply.ts:316]: I'd have squashed this commit into the [previous one](https://github.com/ministryofjustice/hmpps-temporary-accommodation-ui/pull/530/commits/6270416f87a3b44e95c69c68054d45e2baa2840a) given that this commit looks like a fix to get the previous change working.

**Review by tahb — COMMENTED** [review]
Nice work Jerome 🚀  Small bit of feedback from me about git history for next time but otherwise 🤌

**Pull request merged**